cmd/cloud/deployment.go

```diff
@@ -360,3 +371,50 @@ func deploymentVariableUpdate(cmd *cobra.Command, args []string, out io.Writer)
 
 	return deployment.VariableModify(deploymentID, variableKey, variableValue, ws, envFile, variableList, useEnvFile, makeSecret, true, astroClient, out)
 }
+
+func newDeploymentWorkerQueueRootCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "worker-queue",
+		Aliases: []string{"wq"},
+		Short:   "Manage deployment worker queues",
+		Long:    "Manage worker queues for an Astro Deployment.",
+	}
+	cmd.AddCommand(
+		newDeploymentWorkerQueueCreateCmd(out),
+	)
+	return cmd
+}
+
+func newDeploymentWorkerQueueCreateCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"cr"},
+		Short:   "Create a Deployment's worker queue",
+		Long:    "Create a worker queue for an Astro Deployment",
+		Example: "", // TODO should we add some examples for default and custom worker-queue commands?
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deploymentWorkerQueueCreate(cmd, args, out)
+		},
+	}
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "The deployment where the worker queue should be created.")
+	cmd.Flags().StringVarP(&wQueueName, "name", "n", "", "The name of the worker queue. If the name contains a space, specify the entire name within quotes \"\" .")
```

Comment:
We don't allow queue names to have spaces, they actually have to go through regex validation: 
```
export function validateQueueName(name, isDefault) {
  // Shouldn't be reached due to GraphQL schema constraint
  if (isEmpty(name)) {
    throw new Error(`No name provided for worker queue`);
  }
  const nameConstraint = /[a-z]([-a-z0-9]*[a-z0-9])?/;
  // Check non-default WQ
  if (!name.match(nameConstraint).includes(name) || name.length < 1 || name.length > 63) {
    throw new Error(
      `Non-default worker queue must contain only lowercase alphanumeric characters or '-' and start with an alphabetical character. Queue names must not exceed 63 characters.
      `,
    );
  }

  if (name === 'default' && !isDefault) {
    throw new Error(`Non-default worker queue cannot use reserved name: default.`);
  }
  // reserved name due to constraint defined in the AF operator, the operator expects us to pass down a string 'default'
  if (isDefault && name !== 'default') {
    throw new Error(`Default worker queue needs to be named: default`);
  }
}
```

Reply:
@frank-ye Thanks for sharing the name validation logic. I have changed the user facing help message.